### PR TITLE
Persona - onRenderCoin prop

### LINKED ISSUE #1094: Persona: Make a way to customize the Image component
#### Describe the issue:

Currently, the Persona is [integrating](https://github.com/OfficeDev/office-ui-fabric-react/blob/7030f284c3684d6012d8ad65abff158235bc9019/packages/office-ui-fabric-react/src/components/Persona/Persona.tsx#L147) the Image component inside it. It could be great if there is a function to inject customized Image render function.

Within the the customized Image render function, I could image detection or hover/click logic into it.

PeoplePicker provides similar functions ([onRenderItem](https://github.com/OfficeDev/office-ui-fabric-react/blob/78e462bc936350fee2806ee0e70d523e8deee9b5/packages/office-ui-fabric-react/src/components/pickers/PeoplePicker/PeoplePicker.tsx#L26)/[onRenderSuggestionsItem](https://github.com/OfficeDev/office-ui-fabric-react/blob/78e462bc936350fee2806ee0e70d523e8deee9b5/packages/office-ui-fabric-react/src/components/pickers/PeoplePicker/PeoplePicker.tsx#L27)) to inject customized item into the component.

### PR DESCRIPTION
**Pull request by oengusmacinog**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #1094 
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Added an onRenderCoin prop to Persona to allow for passing in or JSX for expanded functionality. Pretty basic set up right now and I am open to suggestions on how to polish this into a solid feature of this component.

#### Focus areas to test

Open to suggestions

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx [hunk 1]
```diff
@@ -143,6 +134,44 @@ export class PersonaCoin extends React.Component<IPersonaProps, IPersonaState> {
     );
   }
 
+  @autobind
+  private _onRenderCoin(props: IPersonaProps, size: number): JSX.Element {
+    if (props.onRenderCoin) {
+      console.log(size);
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: Probably wasn't meant to be included in the commit.

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx [hunk 2]
```diff
@@ -143,6 +134,44 @@ export class PersonaCoin extends React.Component<IPersonaProps, IPersonaState> {
...
+      return(
+        <div>
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: Does the Coin need to be wrapped in `div`s?

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx (hunk 1 above)
Comment (oengusmacinog) [reply packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: Yes, I'll clean that up. I was attempting to make a slightly better example to show. If you have any ideas on that let me know.

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx (hunk 2 above)
Comment (oengusmacinog) [reply packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: I think the change I just committed is a bit cleaner.  The div can be removed if a null return from _onRenderCoin is allowed which would just print nothing.

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx [hunk 3]
```diff
@@ -114,17 +115,7 @@ export class PersonaCoin extends React.Component<IPersonaProps, IPersonaState> {
...
-              imageFit={ ImageFit.cover }
-              src={ imageUrl }
-              width={ coinSize || SIZE_TO_PIXELS[size] }
-              height={ coinSize || SIZE_TO_PIXELS[size] }
-              alt={ imageAlt }
-              shouldFadeIn={ imageShouldFadeIn }
-              shouldStartVisible={ imageShouldStartVisible }
-              onLoadingStateChange={ this._onPhotoLoadingStateChange }
-            />
+            { this._onRenderCoin( this.props, size ) }
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: `onRenderCoin` could be refactored a bit. `IRenderFunction` should take `props` and the default render as parameters. See [Tooltip](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Tooltip/Tooltip.tsx)'s `onRenderContent` for example

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx [hunk 4]
```diff
@@ -5,7 +5,8 @@ import {
   divProperties,
   getInitials,
   getNativeProps,
-  getRTL
+  getRTL,
+  IRenderFunction
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: `IRenderFunction` isn't used

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx (hunk 3 above)
Comment (oengusmacinog) [reply packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: Ah nice! yeah that's much cleaner thank you so much!

**Comment by battletoilet** [comment]
Please run `npm run change` and commit that change file.

File: packages/office-ui-fabric-react/src/components/Persona/examples/PersonaExample.scss [hunk 5]
```diff
@@ -10,4 +10,7 @@
   .example-label {
     margin: 10px 0;
   }
+  .custom-example-coin img {
```
Comment (battletoilet) [inline packages/office-ui-fabric-react/src/components/Persona/examples/PersonaExample.scss]: We don't typically use the kebab-case naming convention for our classes, can you change this one and the one above to camelCase?

**Comment by mikewheaton** [comment]
As @jordandrako pointed out, the Screener regressions are on the master branch and are unrelated to this change.

**Pull request merged**